### DISCUSSION TITLE
ENOENT: no such file or directory, scandir '/Users/ettinger/Library/MobileDevice/Provisioning Profiles/'

### DISCUSSION THREAD

**Discussion by ralyodio** [NativeScript/NativeScript#10188] [General]
How do I deploy this.
I tried Transporter app but it doesn't accept a .app file, only .ipa. 

I saw that I had to run this command:

`tns build ios --release`

but then i get this error:

`ENOENT: no such file or directory, scandir '/Users/ettinger/Library/MobileDevice/Provisioning Profiles/'`

**Comment by rigor789** [comment]
Thank apple for complicating app signing beyond reasonable...

1. make sure you have a valid Apple developer license
2. make sure you have valid certificates/provisioning profiles for your app (and you have registered the bundle id on their developer portal)

Try `ns prepare ios --release` then open `platforms/ios/<appname>.xcworkspace` in XCode, and check the Signing details, if anything is missing - this will allow you to generate a new cert/provisioning profile right in XCode. If that checks out, try `Product -> Archive` within XCode and see if that succeeds. You can upload this directly to TestFlight for release after archiving from the window that pops up (Organizer).

If all went well through XCode, then the CLI command should work as well
```bash
ns build ios --release --for-device
```

**Comment by ralyodio** [comment]
It doesn't create an .ipa file which Teleporter needs.

**Reply by rigor789** [reply]
The XCode archive should produce an `.ipa` - and `ns build ios --release --for-device` as well - the key is the `--for-device` flag, that forces it to build for distribution/devices.

Feel free to join our Discord for help: https://nativescript.org/discord
